UdpDebugLog/src/SphereUdpLogSender/udplog.c

```diff
@@ -0,0 +1,84 @@
+/* Copyright (c) Microsoft Corporation. All rights reserved.
+   Licensed under the MIT License. */
+
+#include "udplog.h"
+#include <stdbool.h>
+#include <stdint.h>
+
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <arpa/inet.h>
+#include <unistd.h>
+
+#include <string.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <stdarg.h>
+
+#define PORT 1824
+
+static int 	sock=-1;
+
+static struct sockaddr_in broadcast_addr;
+static socklen_t addr_len;
+static int ret;
+static char sock_buffer[2048];
+
+static bool slogInit = false;
+
+void initSlog(void);
+
+#ifdef USE_SOCKET_LOG
+int Log_Debug(const char *fmt, ...)
+{
+	if (!slogInit)
+	{
+		initSlog();
+	}
+
+	memset(sock_buffer, 0x00, 2048);
+	va_list args;
+	va_start(args, fmt);
+
+	// You could use the first four bytes to contain a unique value per device.
+	sock_buffer[3] = 0xff;
+	sock_buffer[2] = 0xff;
+	sock_buffer[1] = 0xff;
+	sock_buffer[0] = 0xff;
+
+	vsnprintf(sock_buffer+4, 2048, fmt, args);
```

Comment:
You need to check the return to ensure the buffer is null terminated.

Reply:
And probably handle truncation in some way

Reply:
vsnprintf will truncate and include the null terminator for the space available in the buffer. I can check the return value < 0 for formatting errors and 'perror' on that, and warn through perror that the return from vsnprintf was larger than the input buffer.

Reply:
Ah yes, you're right.  I was thinking of strncpy which can leave you without a null terminator.